lua/java/api/runner.lua

```diff
@@ -0,0 +1,91 @@
+local log = require('java.utils.log')
+local async = require('java-core.utils.async').sync
+local get_error_handler = require('java.handlers.error')
+local jdtls = require('java.utils.jdtls')
+local notify = require('java-core.utils.notify')
+
+local DapSetup = require('java-dap.api.setup')
+
+local M = {}
+local RunnerApi = {}
+
+--- @class RunnerApi
+--- @field client LspClient
+--- @field private dap JavaCoreDap
+function RunnerApi:new(args)
+	local o = {
+		client = args.client,
+	}
+
+	o.dap = DapSetup(args.client)
+	setmetatable(o, self)
+	self.__index = self
+	return o
+end
+
+function RunnerApi:get_config()
+	local configs = self.dap:get_dap_config()
+	log.debug('dap configs: ', configs)
+
+	local config_names = {}
+	local config_lookup = {}
+	for _, config in ipairs(configs) do
+		if config.projectName then
```

Comment:
@s1n7ax  Yes, when I tested with a multimodule application, the `dap.get_dap_configs` method returned the following output
```
.nvim-java-core/lua/java-core/ls/clients/jdtls-client.lua:44: vscode.java.resolveMainClass success! response:  { {
    filePath = "/home/atm1020/workspace/maven-multi-module/module1/src/main/java/com/mycompany/maven/Module1Application.java",
    mainClass = "com.mycompany.maven.Module1Application",
    projectName = "module1"
  }, {
    filePath = "/home/atm1020/workspace/maven-multi-module/module2/src/main/java/com/mycompany/maven/Module2Application.java",
    mainClass = "com.mycompany.maven.Module2Application",
    projectName = "module2"
  }, {
    filePath = "/home/atm1020/workspace/maven-multi-module/module3/src/main/java/com/mycompany/maven/Module3Application.java",
    mainClass = "com.mycompany.maven.Module3Application",
    projectName = "module3"
  }, {
    filePath = "/home/atm1020/.cache/nvim/jdtls/workspaces/common-workspace/maven-multi-module/jdt.ls-java-project/src/com/mycompany/maven/Module2Application.java",
    mainClass = "com.mycompany.maven.Module2Application"
  }, {
    filePath = "/home/atm1020/.cache/nvim/jdtls/workspaces/common-workspace/maven-multi-module/jdt.ls-java-project/src/com/mycompany/maven/Module3Application.java",
    mainClass = "com.mycompany.maven.Module3Application"
  } }
```
Currently, I'm handling this issue by checking if the projectName is not nil.

 I forgot to create a bug ticket for this, i think it would be better to solve this in nvim-java-dap project

Reply:
Do you have this sample java project pushed to a repo?

Reply:
I used this one: https://github.com/colinbut/maven-multi-module

Reply:
@s1n7ax Do you think this issue is valid? 
Should I remove the if statement or not?

Reply:
@atm1020 No. It seems jdtls assigns some random project name in the config for single class files no?

```lua
{ {
    mainClass = "Test",
    name = "test_327a5bbc -> Test",
    projectName = "test_327a5bbc",
    request = "launch",
    type = "java"
  } }
```

Reply:
@s1n7ax  I think this relates to the jdtls cache. The "resolved mainClasses" always appear after a file from another module is opened.  Based on the logs, the project path points to the project cache folder, as you see in my previous comments (For example module2 and module3) .